pkg/rtc/participant.go

```diff
@@ -1355,15 +1355,29 @@ func (p *ParticipantImpl) onSubscribedMaxQualityChange(trackID livekit.TrackID,
 		SubscribedQualities: subscribedQualities[0].Qualities, // for compatible with old client
 		SubscribedCodecs:    subscribedQualities,
 	}
+	// get track's layer dimensions
+	layers := make(map[livekit.VideoQuality]livekit.VideoLayer)
+	trackInfo := p.UpTrackManager.GetPublishedTrack(trackID)
+	if trackInfo != nil {
+		for _, layer := range trackInfo.ToProto().Layers {
+			layers[layer.Quality] = *layer
```

Comment:
Can probably store pointer as it is local here. Yeah, small structure. Probably does not matter to copy.